test/test_ea_mu_plus_lambda.py

```diff
@@ -57,3 +57,97 @@ def objective(ind):
+    n_breedings = 5
+    with pytest.raises(ValueError):
+        _ = cgp.ea.MuPlusLambda(n_offsprings, n_breedings, 1)
+
+
+def test_fitness_wrong_return_type(population_params, genome_params):
+    def objective(individual):
+        individual.fitness = 5
+        return individual
+
+    pop = cgp.Population(**population_params, genome_params=genome_params)
+
+    ea = cgp.ea.MuPlusLambda(10, 10, 1)
+    ea.initialize_fitness_parents(pop, objective)
+    with pytest.raises(ValueError):
+        ea.step(pop, objective)
+
+
+def test_fitness_initialization(population_params, genome_params):
+    def objective(individual):
+        individual.fitness = -1.0
+        return individual
+
+    pop = cgp.Population(**population_params, genome_params=genome_params)
+
+    ea = cgp.ea.MuPlusLambda(10, 10, 1)
+    ea.initialize_fitness_parents(pop, objective)
+    assert all([ind.fitness is not None for ind in pop.parents])
+
+
+def test_one_step(population_params, genome_params):
+    def objective(individual):
+        individual.fitness = float(individual.idx)
+        return individual
+
+    pop = cgp.Population(**population_params, genome_params=genome_params)
+    n_offsprings = 10
+    ea = cgp.ea.MuPlusLambda(n_offsprings, 10, 1)
+    ea.initialize_fitness_parents(pop, objective)
+    old_parent_ids = sorted([ind.idx for ind in pop.parents])
+    ea.step(pop, objective)
+    new_parent_ids = sorted([ind.idx for ind in pop.parents])
+    # After one step, the new parent population should have IDs that
+    # are offset from the old parent ids by n_offsprings
+    # This is by construction in this test because the fitness is equal to the id
+    assert all(
+        [new_id == old_id + n_offsprings for new_id, old_id in zip(new_parent_ids, old_parent_ids)]
+    )
+
+
+def test_fitness_sort(population_params, genome_params):
+    def objective(individual):
+        individual.fitness = float(individual.idx)
+        return individual
+
+    pop = cgp.Population(**population_params, genome_params=genome_params)
+    n_offsprings = 10
+    ea = cgp.ea.MuPlusLambda(n_offsprings, 10, 1)
+    ea.initialize_fitness_parents(pop, objective)
+    sorted_parents = ea._sort(pop.parents)
+    # Assert that the sorting inverted the list of parents (because the fitness is equal to the id)
+    assert sorted_parents == pop.parents[::-1]
+
+
+def test_create_new_populations(population_params, genome_params):
+    def objective(individual):
+        individual.fitness = float(individual.idx)
+        return individual
+
+    pop = cgp.Population(**population_params, genome_params=genome_params)
+    n_offsprings = 10
+    ea = cgp.ea.MuPlusLambda(n_offsprings, 10, 1)
+
+    ea.initialize_fitness_parents(pop, objective)
+
+    offsprings = ea._create_new_offspring_generation(pop)
+    assert len(offsprings) == n_offsprings
+    assert all([ind.idx >= pop.n_parents for ind in offsprings])
+    # Assert that all offspring dna are different from all parents dna
```

Comment:
does this check for "not identity" or "not equality"? in other words, does this make sure that references are not shared between parents and offsprings or that mutations have occurred?

Reply:
This checks that all offspring have different dna, i.e. that they have been mutated.